### PR TITLE
SNOW-179781: Integrate log analyze python standalone repo

### PR DESCRIPTION
**Pull request by sfc-gh-wshangguan**
Add log analyze into this repo.

What this pr did.

1. add `ci/log_analyze_setup.sh`
2. change test password and create a known_ssm_file which contains the new password for log analyze module to use.
3. create `parameters_(aws|azure|gcp)_jenkins.py.gpg`, and `jenkins_test_parameters.py.gpg`. The former will override  `parameters_(aws|azure|gcp).py.gpg` when running on jenkins. The latter helps us to record the new password we are going to use.
4. config pytest to use the right log dir.

### CODE REVIEW

**Comment by sfc-gh-wshangguan** [comment]
https://jenkins.int.snowflakecomputing.com/view/Python%20Connector/job/BuildPyConnector/33/
https://jenkins.int.snowflakecomputing.com/job/RT-PyConnector35/32/
https://jenkins.int.snowflakecomputing.com/job/RT-FIPS-PyConnector36/23/
https://jenkins.int.snowflakecomputing.com/job/RT-PyConnector36/23/
https://jenkins.int.snowflakecomputing.com/job/RT-PyConnector37/23/
https://jenkins.int.snowflakecomputing.com/job/RT-PyConnector38/23/

File: ci/change_snowflake_test_pwd.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        'host': '<host>',
+        'port': '443',
+    }
+
+    # do we need to set time zone?
+    # import time
+    # os.environ['TZ'] = 'UTC'
+    # if not IS_WINDOWS:
+    #     time.tzset()
+
```
Comment (sfc-gh-mkeller) [inline ci/change_snowflake_test_pwd.py:32]:
```suggestion
```

File: ci/change_snowflake_test_pwd.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    conn = snowflake.connector.connect(**params)
+    conn.cursor().execute("use role accountadmin")
+    cmd = "alter user set password = '{}'".format(SNOWFLAKE_TEST_PASSWORD_NEW)
+    print(cmd)
+    conn.cursor().execute(cmd)
+    conn.close()
+
+
+def generate_known_ssm_file():
+    # generate ssm file
```
Comment (sfc-gh-mkeller) [inline ci/change_snowflake_test_pwd.py:45]:
```suggestion
```

File: ci/log_analyze_setup.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+export CLIENT_KNOWN_SSM_FILE_PATH_DOCKER=$CLIENT_LOG_DIR_PATH_DOCKER/rt_jenkins_log_known_ssm.txt
+export CLIENT_KNOWN_SSM_FILE_PATH=$CLIENT_LOG_DIR_PATH/rt_jenkins_log_known_ssm.txt
+echo "[INFO] CLIENT_KNOWN_SSM_FILE_PATH=$CLIENT_KNOWN_SSM_FILE_PATH"
+echo "[INFO] CLIENT_KNOWN_SSM_FILE_PATH_DOCKER=$CLIENT_KNOWN_SSM_FILE_PATH_DOCKER"
+
+# [required envs]
+# To close log analyze, just set ENABLE_CLIENT_LOG_ANALYZE to not "true", e.g. "false".
+export ENABLE_CLIENT_LOG_ANALYZE="true"
+# export ENABLE_CLIENT_LOG_ANALYZE="false"
```
Comment (sfc-gh-mkeller) [inline ci/log_analyze_setup.sh:21]:
```suggestion
```

File: test/conftest.py [hunk 4]
```diff
@@ -69,17 +69,18 @@ def help():
...
-    # TODO: maybe we can use env variable to control whether doing this or not (this should be done)
     # A workround to use our custom Formatter in pytest.
     # Based on the discussion here 'https://github.com/pytest-dev/pytest/issues/2987'
     from snowflake.connector.secret_detector import SecretDetector
     import logging
     import pathlib
     # the directory of this conftest file
-    this_dir = pathlib.Path(__file__).parent.absolute()
+    this_dir = str(pathlib.Path(__file__).parent.absolute())
+    log_dir = os.getenv('CLIENT_LOG_DIR_PATH_DOCKER')
```
Comment (sfc-gh-mkeller) [inline test/conftest.py:79]: It works the way it is, but if you do this you can get rid of the in-line if statement and the previous line too.

```suggestion
    log_dir = os.getenv('CLIENT_LOG_DIR_PATH_DOCKER', str(pathlib.Path(__file__).parent.absolute()))
```

Comment (sfc-gh-wshangguan) [reply test/conftest.py:79]: Perfect point!

**Pull request merged**